FatturaElettronica.Extensions/FatturaElettronicaSignedFileExtension.cs

```diff
@@ -43,5 +47,71 @@ public static void ReadXmlSigned(this Fattura fattura, string filePath)
                 fattura.ReadXml(r);
             }
         }
+
+        public static void SignXml(this Fattura fattura, string pfxFile,string pfxPassword, string p7mFilePath)
+        {
+            if (!File.Exists(pfxFile))
+                throw new FatturaElettronicaSignatureException(Resources.ErrorMessages.PfxIsMissing);
+
+            var cert = new X509Certificate2(pfxFile, pfxPassword);
+            SignXml(fattura, cert, p7mFilePath);
+        }
+        public static void SignXml(this Fattura fattura, X509Certificate2 cert, string p7mFilePath)
+        {
+            string res = string.Empty;
+            string tempFile = Path.GetTempFileName();
+
+            if (!p7mFilePath.ToLowerInvariant().EndsWith(".p7m"))
+                p7mFilePath += ".p7m";
+
+            try
+            {
+                fattura.WriteXml(tempFile);
+
+                ContentInfo content = new ContentInfo(File.ReadAllBytes(tempFile));
```

Comment:
Su Mac questa linea solleva eccezione `System.Security.Cryptography.CryptographicException`: `The OID value is invalid`. Unico riferimento che ho potuto trovare è questo: https://github.com/aspnet/KestrelHttpServer/pull/1940#issuecomment-313565421 dove per altro il codice è diverso, ma spiega cosa potrebbe succedere.

Reply:
Ho provato a cambiare l'inizializzazione degli Oid. Da verificare se risolve il problema su Mac